micro-rdk/src/common/conn/server.rs

```diff
@@ -241,36 +241,32 @@ where
         mut self,
         config: &ConfigResponse,
     ) -> Result<ViamServer<C, T, CC, D, L, NetworkType>, ServerError> {
-        let cfg: RobotCloudConfig = config
-            .config
-            .as_ref()
-            .unwrap()
-            .cloud
-            .as_ref()
-            .unwrap()
-            .into();
-
-        self.mdns
-            .set_hostname(&cfg.name)
-            .map_err(|e| ServerError::Other(e.into()))?;
-        self.mdns
-            .add_service(
-                &cfg.local_fqdn.replace('.', "-"),
-                "_rpc",
-                "_tcp",
-                self.port,
-                &[("grpc", "")],
-            )
-            .map_err(|e| ServerError::Other(e.into()))?;
-        self.mdns
-            .add_service(
-                &cfg.fqdn.replace('.', "-"),
-                "_rpc",
-                "_tcp",
-                self.port,
-                &[("grpc", "")],
-            )
-            .map_err(|e| ServerError::Other(e.into()))?;
+        if let Some(cloud_config) = config.config.clone().unwrap_or_default().cloud {
```

Comment:
there is no need to clone here refs should be enough

Reply:
also not sure if default is the right choice here we should just be able to look if cloud_config exists and if so use a referenceto it